File: 1.py
# Runtime: 36 ms, faster than 99.74% of Python3 online submissions for Two Sum.
# Difficulty: Easy

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        stored_values = dict()
        for index in range(len(nums)):
            diff = target - nums[index]
            if diff in stored_values:
                return [stored_values[diff], index]
            stored_values[nums[index]] = index

File: 1.java
// Runtime: 4 ms, faster than 96.28% of Java online submissions for Two Sum.
// Difficulty: Easy

import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> saved = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            if(saved.containsKey(nums[i])) {
                return new int[]{saved.get(nums[i]), i};
            }
            saved.put(target - nums[i], i);
        }
        return new int[]{-1, -1};
    }
}
